rust/hyperlane-core/src/types/primitive_types.rs

```diff
@@ -0,0 +1,338 @@
+// Based on https://github.com/paritytech/parity-common/blob/a5ef7308d6986e62431e35d3156fed0a7a585d39/primitive-types/src/lib.rs
+
+#![allow(clippy::assign_op_pattern)]
+#![allow(clippy::reversed_empty_ranges)]
+
+use std::fmt::Formatter;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use fixed_hash::{construct_fixed_hash, impl_fixed_hash_conversions};
+use serde::de::Visitor;
+// use serde::ser::SerializeTupleStruct;
```

Comment:
Seems like here and throughout this file there are some commented out bits that we may want to remove?